src/gui/guiFormSpecMenu.cpp

```diff
@@ -971,22 +1079,34 @@ void GUIFormSpecMenu::parseBackground(parserData* data, const std::string &eleme
 			258 + m_fields.size()
 		);
 
-		core::rect<s32> rect;
+		GUIBackgroundImage *e = nullptr;
 		if (!clip) {
 			// no auto_clip => position like normal image
-			rect = core::rect<s32>(pos, pos + geom);
-		} else {
+			core::rect<s32> rect(pos, pos + geom);
```

Comment:
Please change this to use the old design, where you set variables then only create the GUI element in one place. This makes it *a lot* easier to read and reason about. You can set the clip variable in these if-statements

Reply:
Done.